music-rpc.js

```diff
@@ -53,23 +72,32 @@ async function setActivity() {
     switch (state) {
       case 'playing':
         const props = await getProps();
+        const artwork = await searchSong(props);
         console.log('props:', props);
-        rpc.setActivity({
+        const activity = {
           details: props.name,
           state:
             `${props.artist} — ${props.album}` +
             (props.year ? ` (${props.year})` : ''),
           endTimestamp: Math.ceil(
             Date.now() + (props.duration - props.playerPosition) * 1000
           ),
-          largeImageKey: 'appicon',
-          largeImageText: 'Apple Music for macOS',
+          largeImageKey: artwork[0],
+          largeImageText: 'Apple Music',
           smallImageKey: state,
-          smallImageText:
-            `${state[0].toUpperCase() + state.slice(1)}『${props.name}』by ${
-              props.artist
-            }` + (props.year ? ` (${props.year})` : ''),
-        });
+          smallImageText: `${state[0].toUpperCase() + state.slice(1)}`,
+        };
+
+        if (!artwork[1]) {
```

Comment:
```suggestion
        if (artwork[1] !== null) {
```
`!artwork[1]` was not enough...

Reply:
Bruuuuuh js 😩

Reply:
Yeah I think I'll port the code to typescript soon...

Reply:
Well I wanted to do it but no time rn